### PR TITLE
Generate types with DataClassJsonMixin

### PR DESCRIPTION
**Pull request by nathunsmitty**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ PROTOCOL_PATH=$(GOPATH)/src/github.com/keybase/client/protocol
 AVDLC=$(PROTOCOL_PATH)/node_modules/.bin/avdlc
 
 test:
-	poetry run python -m pytest && flake8 && black . && isort -rc .
+	poetry run python -m pytest && poetry run flake8 && poetry run  black . && poetry run isort -rc .
```
Comment (nathunsmitty) [inline Makefile]: Uses the poetry local versions for these tools

**Comment by nathunsmitty** [comment]
@xgess doesn't travis currently run `make test`? Actually, how does that work if mypy was failing previously? Do we return the status code of the commands in `test` correctly?

**Comment by xgess** [comment]
@nathunsmitty we aren't currently running mypy because it failed. we'd need to add it to make test. status codes: i'm not sure. i just assume travis handles all of this correctly, but it's worth checking out.

File: Makefile (hunk 1 above)
Comment (xgess) [reply Makefile]: i dont know if this works...
```suggestion
	poetry run mypy && poetry run python -m pytest && poetry run flake8 && poetry run  black . && poetry run isort -rc . 
```

**Pull request merged**